### DISCUSSION TITLE
Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html"

### DISCUSSION THREAD

**Discussion by abiralEd** [vitejs/vite-plugin-react#351] [Q&A]
I am deploying my react app in AWS S3 bucket. 
Everything works fine locally but each time a new build is deployed, then i get the error as :
[Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html"](https://stackoverflow.com/questions/72070748/failed-to-load-module-script-expected-a-javascript-module-script-but-the-server) in **Chrome** 
and 
Loading failed for the module with source “https://dev.[myappwebsitename].com/assets/index-9af34c50.js”. in firefox

Here is my vite.config.ts"

```
import { defineConfig } from "vite";
import react from "@vitejs/plugin-react";
import path from "path";

export default defineConfig({
  base: '/',

  plugins: [react()],
  server: {
    host: true,
    port: 5173,
  },
  resolve: {
    alias: {
      "@": path.resolve(__dirname, "./src"),
    },
  },
  build: {
    outDir: "dist",
  },
});
```
Here is my package.json scripts:

```

  "scripts": {
    "dev": "vite",
    "host:": "vite --host",
    "build": "tsc && vite build --base=/",
    "build:stage": "tsc && cross-env VITE_APP_ENV=staging vite build",
    "lint": "eslint src --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
    "preview": "vite preview"
  },
```

I have tried following solutions:
1. setting vite.config.ts base to "/" or "./"
2. Disabling brower cache with following code in index.html
 <meta
      http-equiv="Cache-Control"
      content="no-cache, no-store, must-revalidate"
    />
    
  Please review and provide possible solutions.

**Comment by ArnaudBarre** [comment]
You probably get an html file because the file was not found, maybe your not correctly serving the dist folder?
